arch/configure_reader.py

```diff
@@ -0,0 +1,617 @@
+                                                    ARCH_LOCAL=stanza.kvPairs_["ARCH_LOCAL"],
+                                                    BYTESWAPIO=stanza.kvPairs_["BYTESWAPIO"],
+                                                    CFLAGS_LOCAL=stanza.kvPairs_["CFLAGS_LOCAL"],
+                                                    DM_CC=stanza.kvPairs_["DM_CC"],
+                                                    DM_FC=stanza.kvPairs_["DM_FC"],
+                                                    DM_FC_FLAGS=stanza.kvPairs_["DM_FC_FLAGS"],
+                                                    DM_CC_FLAGS=stanza.kvPairs_["DM_CC_FLAGS"],
+                                                    FCBASEOPTS=stanza.kvPairs_["FCBASEOPTS"],
+                                                    FCDEBUG=stanza.kvPairs_["FCDEBUG"],
+                                                    FCNOOPT=stanza.kvPairs_["FCNOOPT"],
+                                                    FCOPTIM=stanza.kvPairs_["FCOPTIM"],
+                                                    M4_FLAGS=stanza.kvPairs_["M4_FLAGS"],
+                                                    SCC=stanza.kvPairs_["SCC"],
+                                                    SFC=stanza.kvPairs_["SFC"],
+                                                    SCC_FLAGS=stanza.kvPairs_["SCC_FLAGS"],
+                                                    SFC_FLAGS=stanza.kvPairs_["SFC_FLAGS"],
+                                                    CPP=stanza.kvPairs_["CPP"],
+                                                    CPP_FLAGS=stanza.kvPairs_["CPP_FLAGS"],
+                                                    )
+
+  # Extra stufff not from stanza but options
+  fmtOption = "set( {opt:<32} {value:<12} CACHE STRING \"Set by configuration\" FORCE )"
+  configStanza += "\n" + "\n".join( [ fmtOption.format( opt=key, value=value ) for key, value in optionsDict.items() ] )
+
+  outputFP = open( output, "w" )
+  outputFP.write( configStanza )
+  outputFP.close()
+
+def projectSpecificOptions( options, stanzaCfg ) :
+  coreOption       = getStringOptionSelection( options.sourceCMakeFile, "WRF_CORE_OPTIONS"    )
+  nestingOption    = getStringOptionSelection( options.sourceCMakeFile, "WRF_NESTING_OPTIONS" )
+  caseOption       = getStringOptionSelection( options.sourceCMakeFile, "WRF_CASE_OPTIONS" )
+  
+  # These are yes
+  yesValues    = [ "yes", "y", "true", "1" ]
+
+  ##############################################################################
+  # Decompose the weird way to write the logic for DM/SM 
+  USE_MPI = False
+  if ( stanzaCfg.serialOpt_ or stanzaCfg.smparOpt_ ) and ( stanzaCfg.dmparOpt_ or stanzaCfg.dmsmOpt_ ) :
+    # togglable
+    # we can safely check this since the user would not have been able to select this stanza if it couldn't be disabled
+    if stanzaCfg.dmCompilersAvailable() :
+      useMPI       = input( "[DM] Use MPI?    [Y/n] : " ).lower() in yesValues
```

Comment:
One potential source of confusion for users (it confused me at first!) is that in the script there are several yes/no questions that are posed with [Y/n] at the end of the line. The capital 'Y' typically indicates that this is the default answer, however, the script defaults answers to false because the optional is set using `input().lower() in yesValues`. So pressing return after these questions, assuming the default unanswered value will be 'yes', leads to unexpected results.

Reply:
I genuinely didn't know this! I thought the alternation of capital/lower `Y/n` was meant to indicate to the user that any capital or lower input would be acceptable. This is definitely confusing then.

Reply:
Another option would be to not allow empty input, that would make things more verbose but eliminate any ambiguity.

Reply:
Would swapping to `y/N` be acceptable, or do you think some of these options should be on by default while some off?

Reply:
I think at the very least MPI should default to true given that most non-trivial cases require it. Regarding the others, I'm not sure. Others closer to the code may have a more valid opinion...

Reply:
@mgduda Do you have a particular preference on which way we go with default options like this?

Reply:
I would agree with @rcabell that MPI should probably default to `Y`. I would also suggest that `BASIC` nesting (nesting option `1`) should be the default as well.